### DISCUSSION TITLE
Verify Libre.fm account on Mastodon

### DISCUSSION THREAD

**Discussion by janerationx** [foocorp/hacienda#95] [General]
(If I missed this somehow, sorry.)

Is there any way to verify a Libre.fm account to Mastodon? It seems the way Libre profiles are now, it isn't possible.

**Comment by mattl** [comment]
It works… sometimes. Take a look at my profile https://social.coop/@mattl and you’ll see it worked. Some others have had success. 

It’s not working 100% of the time because of one off the firewall rules in place on Cloudflare. So I do need to fix it.

**Comment by janerationx** [comment]
Okay. Thank you.

**Comment by mattl** [comment]
Try editing your profile to remove the link and re-add it. Let me know if that fixes anything.

**Comment by janerationx** [comment]
Nope. Still unverified on Mastodon.

The problem, I think, is that the Mastodon link on my Libre profile needs to have "rel=me". There's no way to do that currently.

**Comment by mattl** [comment]
It does it for you. that’s what indiewebify checks too

**Comment by janerationx** [comment]
Nope. Still nothing. Oh well. Thanks for trying to help.

**Comment by mattl** [comment]
I’ll take a further look tomorrow. Who hosts your solo instance?

**Comment by janerationx** [comment]
Masto Host.

**Comment by mattl** [comment]
I’ll email Hugo tomorrow

**Comment by janerationx** [comment]
Any updates on this?

**Comment by mattl** [comment]
Not yet, I wanted to try a few more things in my configuration first. Haven't had a chance to get to that yet.

**Comment by alxtrnr** [comment]
It worked with toot.io - https://toot.io/@alxtrnr
It's not worked with newsmast.social - https://newsmast.social/@alxtrnr

**Comment by unseenlarks** [comment]
i'm on mastodon.online, doesn't seem to work there (unless of course, _i_'ve managed to botch it). :sweat_smile:

**Comment by mattl** [comment]
I need to work on a debug URL we can use to work this out.

**Comment by ghost** [comment]
I ran into a character limit with adding my Fediverse url to my Libre.fm profile, it cuts it off at 41 characters or so. My fault for using a subdomain I guess. It might be better to allow us to use our handle (@name@domain.tld)?

**Comment by mattl** [comment]
No it's a URL for a reason. Nobody can do anything with a username by default.

**Comment by ghost** [comment]
I'm not sure if it's a standard across all Fediverse/AP things but couldn't webfinger be used? Excuse my ignorance on the subject. I guess just bumping up the character limit a little would be welcome 🙂

**Comment by mattl** [comment]
I'd like something that every user can make use of. A URL fulfills that requirement. 

I'm not even sure what a WebFinger would look like.

**Comment by janerationx** [comment]
This is such a vexing problem. Is there any solution in sight, or should I close this issue?

**Comment by mattl** [comment]
I need to try and find a way to fix it that won't bring the site down too. 

Basically some of these hosts are also used by bots who take up too many resources or have scripts which try to make 500 accounts in an hour, so we block them.

**Comment by janerationx** [comment]
I don't understand the connection between verifying a URL and bots/scripts, especially since verification works on some Mastodon servers and not others.

**Comment by mattl** [comment]
Let me try and explain it...

So when the Mastodon server verifies things, it looks at your profile and if it finds the link to your Mastodon profile it can verify the link.

Every server has an IP address and IP addresses belong to groups known as Autonomous Systems (AS) -- every company has what's called an AS number, it's a unique number for their network. 

For example Bytemark who host Libre.fm, their number is 35425 -- part of IOMART who bought Bytemark a couple of years ago.

I block some hosting providers entirely because they allow anyone to spin up a server in seconds and hurt Libre.fm for everyone. For example, making loads of fake accounts for spam purposes. 

So if your Mastodon server is hosted by a provider I'm blocking, the lookup will fail because that Mastodon server is blocked from even trying to load a page on Libre.fm.

---

However, all is not lost... I should be able to make an exception for the individual IP address of any Mastodon server. 

@janerationx remind me of your Mastodon server?

**Comment by mattl** [comment]
@janerationx Okay, back at my desk. 

Try editing your profile now, see if you can get it to verify.

**Comment by janerationx** [comment]
My server is hosted by Masto Host. The url is https://janerationx.rocks/@j.

I'll give it a try, see what happens.

**Comment by janerationx** [comment]
It still doesn't verify.

**Comment by mattl** [comment]
@janerationx Please try again now, I found the IP address from Mastohost that was doing the check and added it to the allowlist.

**Comment by janerationx** [comment]
No verification. I don't think this problem can be fixed, so I'm going to close the issue. 

Thank you for trying. I do appreciate it.

**Comment by mattl** [comment]
For anyone having issues getting their verification working on Mastodon, please list your profile below and if you know it, the host of your Mastodon profile (ie. Mastohost) 

https://github.com/foocorp/hacienda/issues/48#issuecomment-2625924446 explains why there some issues.
